utils/results/results.go

```diff
@@ -0,0 +1,315 @@
+}
+
+type TargetResults struct {
+	ScanTarget
+	// All scan results for the target
+	ScaResults *ScaScanResults  `json:"sca_scans,omitempty"`
+	JasResults *JasScansResults `json:"jas_scans,omitempty"`
+	// Errors that occurred during the scans
+	Errors      []error    `json:"errors,omitempty"`
+	errorsMutex sync.Mutex `json:"-"`
+}
+
+type ScaScanResults struct {
+	IsMultipleRootProject *bool `json:"is_multiple_root_project,omitempty"`
+	// Target of the scan
+	Descriptors []string `json:"descriptors,omitempty"`
+	// Sca scan results
+	XrayResults []services.ScanResponse `json:"xray_scan"`
+}
+
+type JasScansResults struct {
+	ApplicabilityScanResults []*sarif.Run `json:"contextual_analysis,omitempty"`
+	SecretsScanResults       []*sarif.Run `json:"secrets,omitempty"`
+	IacScanResults           []*sarif.Run `json:"iac,omitempty"`
+	SastScanResults          []*sarif.Run `json:"sast,omitempty"`
+}
+
+func NewCommandResults(xrayVersion string, entitledForJas bool) *SecurityCommandResults {
+	return &SecurityCommandResults{XrayVersion: xrayVersion, EntitledForJas: entitledForJas, targetsMutex: sync.Mutex{}}
+}
+
+func (r *SecurityCommandResults) SetMultiScanId(multiScanId string) *SecurityCommandResults {
+	r.MultiScanId = multiScanId
+	return r
+}
+
+// --- Aggregated results for all targets ---
+
+func (r *SecurityCommandResults) GetTargetsPaths() (paths []string) {
+	for _, scan := range r.Targets {
+		paths = append(paths, scan.Target)
+	}
+	return
+}
+
+func (r *SecurityCommandResults) GetScaScansXrayResults() (results []services.ScanResponse) {
+	for _, scan := range r.Targets {
+		results = append(results, scan.GetScaScansXrayResults()...)
+	}
+	return
+}
+
+func (r *SecurityCommandResults) GetJasScansResults(scanType jasutils.JasScanType) (results []*sarif.Run) {
+	if !r.EntitledForJas {
+		return
+	}
+	for _, scan := range r.Targets {
+		results = append(results, scan.GetJasScansResults(scanType)...)
+	}
+	return
+}
+
+func (r *SecurityCommandResults) GetErrors() (err error) {
+	err = r.Error
+	for _, target := range r.Targets {
+		for _, targetErr := range target.Errors {
+			err = errors.Join(err, targetErr)
+		}
+	}
+	return
+}
+
+func (r *SecurityCommandResults) GetTechnologies() []techutils.Technology {
+	technologies := datastructures.MakeSet[techutils.Technology]()
+	for _, scan := range r.Targets {
+		technologies.AddElements(scan.GetTechnologies()...)
+	}
+	return technologies.ToSlice()
+}
+
+// In case multipleRoots is true, the field Component will show the root of each impact path, otherwise it will show the root's child.
+// Set multipleRoots to true in case the given vulnerabilities array contains (or may contain) results of several projects or files (like in binary scan).
+func (r *SecurityCommandResults) HasMultipleTargets() bool {
+	if len(r.Targets) > 1 {
+		return true
+	}
+	for _, scanTarget := range r.Targets {
+		// If there is more than one SCA scan target (i.e multiple files with dependencies information)
```

Comment:
more than one? the code indicated one or more, so just fix the comment so it'll be clear

Reply:
should be more than one, fixed in code